### PR TITLE
REP-4857 SAML filter modify response Part1/Take2

### PR DESCRIPTION
**Pull request by wdschei**
This addresses the comments left on #1660.
And cleans up a few things that were found along the way.

### CODE REVIEW

**Comment by wdschei** [comment]
Retest this please.

**Comment by wdschei** [comment]
Trying to force Jenkins to fire.

File: repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala [hunk 1]
```diff
@@ -188,11 +186,15 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
...
+    var documentBuilder : DocumentBuilder = null
     try {
-      documentBuilderFactory.newDocumentBuilder().parse(samlResponse)
+      documentBuilder = XMLParserPool.borrowParser
+      documentBuilder.parse(samlResponse)
     } catch {
       case se: SAXException =>
         throw SamlPolicyException(SC_BAD_REQUEST, "SAMLResponse was not able to be parsed", se)
+    } finally {
+      if (documentBuilder != null) XMLParserPool.returnParser(documentBuilder)
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala]: No curly on if, i'm surprised the style checker didn't catch that one. I went with an optional for the xpath version of this.

File: repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala [hunk 2]
```diff
@@ -493,7 +495,7 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
     private var initialized = false
 
     override def configurationUpdated(configurationObject: SystemModel): Unit = {
-      sendTraceHeader = Option(configurationObject.getTracingHeader).map(_.isEnabled).getOrElse(true)
+      sendTraceHeader = Option(configurationObject.getTracingHeader).forall(_.isEnabled)
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala]: Unrelated, but i like it.

Comment (wdschei) [inline repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala]: I was clearing all the items in the right hand trough and this is one the IntelliJ static analysis found.

File: repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala (hunk 1 above)
Comment (wdschei) [inline repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala]: I was thinking about doing that. Using the `null` just doesn't seem right. I just based it directly off the example in the original comments:

* https://github.com/rackerlabs/api-checker/blob/master/util/src/test/scala/com/rackspace/com/papi/components/checker/util/XMLParserPoolSuite.scala#L27

Comment (RackerWilliams) [inline repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala]: ...You should take Scala coding advice from @adrianjgeorge  before you take it from me.  Using an optional is totally legit in this case.

**Pull request merged**